src/blockdata/locktime.rs

```diff
@@ -0,0 +1,631 @@
+/// * [BIP-113 Median time-past as endpoint for lock-time calculations](https://github.com/bitcoin/bips/blob/master/bip-0113.mediawiki)
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub enum LockTime {
+    /// A block height lock time value.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use bitcoin::LockTime;
+    ///
+    /// let n_lock_time: u32 = 741521;
+    /// let lock_time = LockTime::from_consensus(n_lock_time);
+    ///
+    /// assert!(lock_time.is_block_height());
+    /// assert_eq!(lock_time.to_consensus_u32(), n_lock_time);
+    ///
+    /// let n = LockTime::from_consensus(741520); // From `741520 OP_CHECKLOCKTIMEVERIFY`.
+    /// assert!(n.is_same_unit(lock_time));
+    /// assert!(n.is_satisfied(lock_time).expect("same time lock units don't error"));
+    /// ```
+    Blocks(Height),
+    /// A UNIX timestamp lock time value.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use bitcoin::LockTime;
+    ///
+    /// let n_lock_time: u32 = 1653195600; // May 22nd, 5am UTC.
+    /// let lock_time = LockTime::from_consensus(n_lock_time);
+    //
+    /// assert!(lock_time.is_block_time());
+    /// assert_eq!(lock_time.to_consensus_u32(), n_lock_time);
+    ///
+    /// let n = LockTime::from_consensus(1653195000); // From `1653195000 OP_CHECKLOCKTIMEVERIFY`.
+    /// assert!(n.is_same_unit(lock_time));
+    /// assert!(n.is_satisfied(lock_time).expect("same time lock units don't error"));
+    /// ```
+    Seconds(Time),
+}
+
+impl LockTime {
+    /// Constructs a `LockTime` from an nLockTime value or the argument to OP_CHEKCLOCKTIMEVERIFY.
+    pub fn from_consensus(n: u32) -> Self {
+        if is_block_height(n) {
+            Self::Blocks(Height::from_consensus(n).expect("n is valid"))
+        } else {
+            Self::Seconds(Time::from_consensus(n).expect("n is valid"))
+        }
+    }
+
+    /// Returns true if both lock times use the same unit i.e., both height based or both time based.
+    pub fn is_same_unit(&self, other: LockTime) -> bool {
+        mem::discriminant(self) == mem::discriminant(&other)
+    }
+
+    /// Returns true if this lock time value is a block height.
+    pub fn is_block_height(&self) -> bool {
+        match *self {
+            LockTime::Blocks(_) => true,
+            LockTime::Seconds(_) => false,
+        }
+    }
+
+    /// Returns true if this lock time value is a block time (UNIX timestamp).
+    pub fn is_block_time(&self) -> bool {
+        !self.is_block_height()
+    }
+
+    /// Returns true if this timelock constraint is satisfied by the respective `height`/`time`.
+    ///
+    /// If `self` is a blockheight based lock then it is checked against `height` and if `self` is a
+    /// blocktime based lock it is checked against `time`.
+    pub fn is_satisfied_by_either(&self, height: Height, time: Time) -> bool {
```

Comment:
This function should be a function in `impl Transaction` along with the disabled check. 

https://github.com/bitcoin/bitcoin/blob/b9122e95f0f4ff5d2b2e21a5caf6c69d488c0347/src/consensus/tx_verify.cpp#L17

Reply:
Thanks! PR now includes `Transaction::is_final`.